### ISSUE TITLE
CloudWatch Logs: Region dropdown doesn't allow template variables

### ISSUE THREAD

**Issue by wjbuys** [grafana/grafana#45429]
I want to use a template variable to set the region for the CloudWatch Logs datasource. However, the UI doesn't allow selecting/entering template variables.

This does seem to be supported when using CloudWatch Metrics, and it even seems to be supported by the CloudWatch Logs JSON model (i.e. the following snippet of panel JSON works):

```json
  // ...
  "type": "logs",
  "datasource": {
    "type": "cloudwatch"
  },
  "targets": [
    {
      "queryMode": "Logs",
      "region": "$Region",
      "datasource": {
        "type": "cloudwatch",
      },
      // ...
    }
```

_Originally posted by @Eraac in https://github.com/grafana/grafana/issues/24333#issuecomment-684599523_

**Comment by zuchka** [comment]
Thanks for creating this issue, @wjbuys 

what version are you using? and specifically what kind of variable? thx 👍

**Comment by wjbuys** [comment]
This is on Grafana 8.3.4 (but I think it is still not fixed in 8.3.6), and it's a Custom variable (though ideally this should work with a Query variable too)

**Comment by iwysiu** [comment]
Hi @wjbuys ! This will be supported by the new query header added in #44668 . For older versions, like you mentioned, it can be worked around by editing the panel json.
